airflow/jobs/scheduler_job.py

```diff
@@ -1290,7 +1256,7 @@ def _enqueue_task_instances_with_queued_state(self, simple_dag_bag,
         """
         TI = models.TaskInstance
         # actually enqueue them
-        for simple_task_instance in simple_task_instances:
+        for i, simple_task_instance in enumerate(simple_task_instances):
```

Comment:
Why do we need enumerate if the ``i`` variable  is not used inside the loop?

Reply:
Used by the `return` -- basically this avoids iterating over the loop, and then counting the length of the loop again.

_This_ might have been why I wanted to avoid the `len()` call, in my large dags there are 1000s of dags. But I suspect this was me just making stuff up that isn't really needed :D

Reply:
Yeah no, this is not needed. Reverthing this change.